new/property.md

```diff
@@ -0,0 +1,223 @@
+
+We will need to write a comprehensive resource writing guide as well, in order to get it pumped to the top of Google, so that `attribute` comes up less and less often in searches and `property` comes up more and more.
+
+### Property type
+
+Properties with a single type are common enough that we support a "type" for a property, specifiable after its name.
+
+```ruby
+class MyResource < Chef::Resource
+  property :content, String
+end
+```
+
+This is actually an alias for `is` (described later here).
+
+#### Reusing types
+
+The type of a property is represented by `Chef::Resource::PropertyType`, and contains accessors for all the properties of a type (`must_be`, `name_attribute`, `kind_of`, etc.).
+
+When you declare `property :name, <type>, <options>`, one of two things happens:
+
+- If the type is a PropertyType instance, it is dup'd and any <option>s are set on the new type.
+- If the type is a Class, a new PropertyType instance is created with `kind_of` set to [Class] and any options are set on the new type.
+- If type is not passed, a new PropertyType instance is created, and any options are set on the new type.
+
+#### Property Inheritance
+
+Subclasses get properties from their parent.
+
+```ruby
+class A < Chef::Resource
+  property :a, String
+end
+class B < Chef::Resource
+  property :b, String
+end
+B.state_attrs -> [ :a, :b ]
+```
+
+#### Overriding Properties
+
+When a property is overridden, the override is *complete*: that is, the parent
+type is not extended or mixed in any way.
+
+```ruby
+class A < Chef::Resource
+  property :a, String, default: 'Hello'
+end
+class B < A
+  property :a
+end
+A.properties[:a].default #=> 'Hello'
+B.properties[:a].default #=> nil
+```
+
+#### `is`
+
+`is` is a new validation parameter that uses Ruby's match operator `===` (the thing that drives `case` and `when`).
+
+```ruby
+# These are equivalent
+property :x, [ :a, :b, :c ]
+property :x, is: [ :a, :b, :c ]
+```
+
+It is worth noting that many existing validations can be expressed directly in terms of `is`:
+
+```ruby
+# These:
+attribute :path, kind_of: String
+attribute :path, equal_to: [ :a, :b, :c ]
+attribute :path, regex: /^\//
+attribute :path, respond_to: :merge
+attribute :path, cannot_be: :empty
+property :path, is: String
+property :path, is: [ :a, :b, :c ]
+property :path, is: /^\//
+property :path, is: proc { |v| v.respond_to?(:merge) }
+property :path, is: proc { |v| !v.empty? }
+```
+
+As well as some things that were hard to express before:
+
+```ruby
+property :path, [ String, :up, :down, nil ]
+```
+
+If both `is` and a type are specified, the values in the type are prepended to `is`.
+
+### RSpec matchers
+
+`is` allows for rspec 3 matchers to be passed, and will validate them and print failures.
+
+```ruby
+include RSpec::Matchers
+property :path, a_string_starting_with('/')
```

Comment:
I actually wasn't planning on including them natively actually--just for people who wanted, it becomes an immediate option (since RSpec Matchers implement `===`).